lib/s3-s3.js

```diff
@@ -40,9 +46,25 @@ var s3 = function (primaryS3, secondaryS3) {
           setupOn(realRequest, 'extractError', onFunc);
           setupOn(realRequest, 'extractData', onFunc);
           setupOn(realRequest, 'success', onFunc);
-          setupOn(realRequest, 'error', onFunc);
           setupOn(realRequest, 'complete', onFunc);
 
+          // error is a special on() call which we use to see if this is a call that needs to
+          // go to the secondary
+          realRequest.on('error', function(error) {
+            if (usingPrimary && error &&
+                error.hasOwnProperty('code') &&
+              (error.code === 400 || error.code >= 500)) {
```

Comment:
I am not yet confident this is the right set up conditions to failover.  There's not even good documentation that I could find on the error object- I had to dig through the source to figure out the code property was there.  So we may need to tweak this after some testing.